addons/website_mass_mailing/static/src/js/website_mass_mailing.editor.js

```diff
@@ -28,6 +28,7 @@ options.registry.mailing_list_subscribe = options.Class.extend({
             'id': this.popup_template_id,
             'window_title': this.popup_title,
             'select': _t("Newsletter"),
+            'default': self.$target.attr("data-list-id") !== "0" ? self.$target.attr("data-list-id") : null,
             'init': function (field, dialog) {
                 return rpc.query({
                     model: 'mailing.list',
```

Comment:
I wanted to do something like that at first but what about the 'default' option in utils.js, is it not supposed to be there for this purpose ?

And is it normal that by given a default value, this line
`field.val(options['default']);`  (file addons/website/static/src/js/utils.js, line 142 )
don't seem to do the job anyway? (the reason why I put it again after init )

What do you think? Thanks

Reply:
The .val() at line 142 doesn't work because it's trying to set a value when the select field is not even filled with the options yet.
Looking at the code, I don't think that "utils" supports setting values for select fields at all.

Your change is good, but it's not enough, we would also have to support select fields that are not filled by a promise, and also probably write some tests, ...

That's why I prefer a "simple" fix for our use case only, rather than a generic fix that will lead to more code / more tests / more review.

(But again, it was a good idea to attempt a generic fix, it's just not fitting for this use case in particular).

Reply:
Alright, Thank you very much for the explanation.